### DISCUSSION TITLE
New to yarn. Help setting up for contributing

### DISCUSSION THREAD

**Discussion by dtressel** [tonaljs/tonal#416] [Q&A]
Hi all! This is the first time having to deal with Yarn as a package manager, and I'd like to confirm what is the best way to set up everything for contributing. I'm working on Windows by the way. On the Yarn website, it says that the recent versions of Yarn should be installed locally and not globally. Two quick questions:

1. Is there a particular version of Yarn that I should be using?
2. For this project is it best to use a globally or locally installed Yarn?

**Comment by danigb** [comment]
Hi,

You're right, using yarn makes more difficult to contribute to this project. I'll move to npm. The reason to use yarn was because when this project started, monorepos (one repository with multiple packages) wasn't supported on npm. But that changed long time ago.

EDIT: From now on, we're using npm - https://github.com/tonaljs/tonal/pull/417

**Reply by dtressel** [reply]
Hi!

Thanks for moving to npm to make it easier to contribute! I'm new to contributing to open source so some of my struggles are due to that. I'm close to setting things up now that you've switched to npm, but there's still some things going on. I've tried to follow the steps in the contributing.md doc and ran into a few problems. What I've done:

1. cloned the repo
2. navigated to tonal folder and ran "npm install"
3. "npm build" didn't work so I ran "npm run build"
4. Everything looked good so I decided to try to run tests
5. "npm test:ci" didn't work so I ran "npm run test:ci"
6. "npm run test:ci" resulted in the following error:
```
> lint
> eslint 'packages/**/*.ts'

Oops! Something went wrong! :(

ESLint: 8.56.0

No files matching the pattern "'packages/**/*.ts'" were found.
Please check for typing mistakes in the pattern.
```

7. Running the tests on their own worked fine, but using the "test:ci" script failed because "npm run lint" is failing due to the above.
8. One other weird thing is that git is detecting changes after running "npm test:ci" in 107 files even though there no actual changes in any of the files. Maybe this has something to do with the linter...?

Anyway, let me know if you have any thoughts. 

All the best,
Daniel

**Comment by danigb** [comment]
Hey! Thanks for the detailed response. All I can say is that every step works for me. It's weird you have git changes. I'd first remove all those changes (`git checkout .` maybe?), ensure you're at latest main commit (`git checkout main && git pull` maybe?) and start again all steps (in fact, only `npm run test:ci` is required). 

There are typescript files inside `packages` folder, so something is defenitely wrong here... Not sure what. I'm sorry
